### ISSUE TITLE
gganimate doesn't correctly render text in expression()

### ISSUE THREAD

**Issue by moondog1969** [thomasp85/gganimate#506]
Plotmath text with `expression()` is not working on axes in animations.  For instance, the 2 is not super-scripted for the equation on the y-axis label in the plot below.

````
library(gganimate)

ggplot(mtcars, aes(mpg, disp)) +
  geom_point(aes(color = gear)) +
  transition_states(gear, transition_length = 2, state_length = 1) +
  enter_fade() +
  ylab(expression(E = mc^2)) +
  exit_fade()
````

I'm using:

````
R version 4.4.2 (2024-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 10 x64 (build 17134)
````

gganimate version: `1.0.9`.

This seems to be a new issue, as gganimate axes were using `expression()` correctly until recently.

Thanks

**Comment by moondog1969** [comment]
Will work correctly using `bquote()` instead of `expression()`:

````
e <- bquote(E == mc^2)

ggplot(mtcars, aes(mpg, disp)) +
  geom_point(aes(color = gear)) +
  transition_states(gear, transition_length = 2, state_length = 1) +
  enter_fade() +
  ylab(e) +
  exit_fade()
````

**Comment by aphalo** [comment]
@thomasp85 @teunbrand 
`bquote()` returns a call instead of an expression. The catch is that `bquote()` is no longer supported by 'ggplot2' (3.5.1.9000), tested today. This seems to be as a result of implementing support for lambdas as argument to `names` in scales. Now a `call` object as argument to `name` errors... so `bquote()` cannot be used.

I think both `bquote()` and `expression()` are rather frequently used for axis labels, at least I had both in 'ggpmisc' and 'ggspectra' in different statements. I edited both packages to only use expressions for `name`. However, this approach does not work with 'gganimate'. Ideally, both calls and expressions should continue to be supported.

**Comment by teunbrand** [comment]
@aphalo Should this be an issue for ggplot2 to consider rather than gganimate? Can you give examples of `bquote()` and/or `expression()` that used to work but no longer do?

**Comment by aphalo** [comment]
@teunbrand Yes, I am writing it. It does not seem so simple as I wrote above. `bquote()` seems to still work in simpler cases... or you have already fixed the problem in the last day or so. I will open the issue in 'ggplot2' once I can write a simple reprex. Hopefully, very soon.
